packages/react-button/src/components/Button/Button.types.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { ComponentProps, ShorthandValue, ComposeOptions, ComposeStandardStatics } from '../../temp/tempTypes';
+
+export type SizeValue = 'smallest' | 'smaller' | 'small' | 'medium' | 'large' | 'larger' | 'largest';
+
+// tslint:disable-next-line:no-any
+export interface ButtonProps extends ComponentProps, React.HTMLAttributes<HTMLButtonElement> {
+  /**
+   * Shorthand icon. A shorthand prop can be a literal, object,
+   * JSX, or function which takes render options.
+   */
+  icon?: ShorthandValue<{}>;
+
+  /**
+   * Shorthand content within the button.
+   */
+  content?: ShorthandValue<{}>;
+
+  /**
+   * Shorthand loader content within the button.
+   */
+  loader?: ShorthandValue<{}>;
+
+  /** A button can appear circular. */
+  circular?: boolean;
+
+  /** A button can show that it cannot be interacted with. */
+  disabled?: boolean;
+
+  /** A button can fill the width of its container. */
+  fluid?: boolean;
+
+  /** A button can contain only an icon. */
+  iconOnly?: boolean;
+
+  /** An icon button can format its Icon to appear before or after its content */
+  iconPosition?: 'before' | 'after';
+
+  /** A button that inherits its background and has a subtle appearance */
+  inverted?: boolean;
+
+  /** A button can show a loading indicator. */
+  loading?: boolean;
+
+  /**
+   * Called after a user clicks the button.
+   * @param event - React's original SyntheticEvent.
+   * @param data - All props.
+   */
+  // onClick?: ComponentEventHandler<ButtonProps>;
```

Comment:
Are these commented because you're getting them from the `HTMLAttributes`?

Reply:
Not quite. I didn't understand it at first. I now do, but it's not supported yet.

```tsx
  /**
   * Called after a user clicks the button.
   * @param event - React's original SyntheticEvent.
   * @param data - All props.
   */
  onClick?: ComponentEventHandler<ButtonProps>;
```

Notice the 2nd param, data. This requires we always override `onClick` (and `onFocus`) to provide the 2nd param.

Reply:
It's convenient to be able to access props from the `onClick` but it certainly requires some factory magic to standardize. I'm still a little mixed on it.

Reply:
I'll leave it there, lets come back to this.